Package.swift

```diff
@@ -331,3 +333,20 @@ if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == nil {
         .package(path: "../swift-driver"),
     ]
 }
+
+// SwiftSyntax relies on a matching copy of lib_InternalSwiftSyntaxParser.dylib,
+// which is part of a Swift toolchain (but can also be embedded in an app as a
+// standalone library). To ensure SwiftPM is still easy to build and test
+// independently of a full build-script invocation, only include the PackageSyntax
+// target if SPM_BUILD_PACKAGE_SYNTAX is set. See the SwiftSyntax
+// [README](https://github.com/apple/swift-syntax/blob/main/README.md) for more
+// information about using it as a dependency.
```

Comment:
So because of the conditionality here, clients that use libSwiftPM would continue building because `SPM_BUILD_PACKAGE_SYNTAX` isn't set by default, but the package editing functionality just wouldn't be available?  Is the idea that none of the other code in libSwiftPM would import `PackageSyntax`, or would it use conditional imports to make sure it continues working when `PackageSyntax` isn't available?  I suppose there's some risk that if the tests run with `PackageSyntax` checked out, unintentional imports might slip through that only show up when trying to use libSwiftPM without `PackageSyntax` (or I suppose the self-hosted tests would catch that?).

Reply:
> So because of the conditionality here, clients that use libSwiftPM would continue building because SPM_BUILD_PACKAGE_SYNTAX isn't set by default, but the package editing functionality just wouldn't be available? 

Yeah, it seemed like leaving it off by default was best for incremental adoption by library clients, and to ensure that somebody can still build a standalone checkout of the SwiftPM repo easily.

> Is the idea that none of the other code in libSwiftPM would import PackageSyntax, or would it use conditional imports to make sure it continues working when PackageSyntax isn't available?

I think conditional imports are going to be required in the `Commands` and `CommandsTests` modules to support the CLI, but other than that I'm pretty sure we can avoid them. It's definitely a bit messy, but like you mentioned the self-hosted tests build without `SPM_BUILD_PACKAGE_SYNTAX ` set so they should catch any accidental build breakage.

Reply:
Yes, I absolutely agree that leaving `PackageSyntax` out for now existing clients is the right thing.  And it makes sense to use `#if BUILD_PACKAGE_SYNTAX` for this (as seen in https://github.com/apple/swift-package-manager/pull/3180; I forgot about that part when leaving this comment earlier).